src/resolvers.ts

```diff
@@ -69,39 +75,71 @@ export const resolvers = {
     },
   },
   Mutation: {
-    async createUser(_, { data }, context): Promise<UserOutput> {
+    async createUser(_, { input: userInput }, context): Promise<UserOutput> {
       checkToken(context);
 
       const user = new User();
-      const hashedPassword = passwordHashing(data.password);
+      const hashedPassword = passwordHashing(userInput.password);
 
-      user.name = data.name;
-      user.email = data.email;
+      user.name = userInput.name;
+      user.email = userInput.email;
       user.password = hashedPassword;
-      user.birthDate = data.birthDate;
+      user.birthDate = userInput.birthDate;
 
-      checkPassword(data.password);
-      await checkEmail(data.email);
+      checkPassword(userInput.password);
+      await checkEmail(userInput.email);
 
       await AppDataSource.manager.save(user);
       return user;
     },
-    async login(_, { data }): Promise<LoginOutput> {
+    async login(_, { input: loginInput }): Promise<LoginOutput> {
       const user = await userRepository.findOneBy({
-        email: data.email,
-        password: passwordHashing(data.password),
+        email: loginInput.email,
+        password: passwordHashing(loginInput.password),
       });
 
       if (!user) {
         throw new CustomError('User not found, please create an account, or review credentials', 401);
       }
 
-      const token = generateToken(user.id, data.rememberMe);
+      const token = generateToken(user.id, loginInput.rememberMe);
 
       return {
         user: user,
         token: token,
       };
     },
+    async createAddress(_, { input: addressInput }, context): Promise<AddressOutput> {
+      checkToken(context);
+
+      const user = await userRepository.findOne({
+        where: { email: addressInput.userEmail },
+        relations: { address: true },
+      });
+
+      if (!user) {
+        throw new CustomError('User not present in the database', 404);
+      }
+
+      const address = new Address();
+
+      address.state = addressInput.state;
+      address.city = addressInput.city;
+      address.neighborhood = addressInput.neighborhood;
+      address.street = addressInput.street;
+      address.streetNum = addressInput.streetNum;
+      address.cep = addressInput.cep;
+      address.complement = addressInput.complement;
+
+      address.user = user;
+
+      if (await userRepository.findOneBy({ address: address })) {
+        throw new CustomError('This User is Alredy Registered in this Address', 400);
+      }
+
+      await addressRepository.save(address);
+
+      return address;
```

Comment:
esse `address` que vc retornar tem o id atualizado? o `save` do repository n retorna um valor atualizado?
```suggestion
      return addressRepository.save(address);
```

Reply:
Opa Edu, retornando o address, o id vem atualizado sim: 
![image](https://user-images.githubusercontent.com/89315525/215776742-282a6fe9-2887-44cf-a01c-59266e553b70.png)

E retornando o addressRepository.save(address) tb: 
![image](https://user-images.githubusercontent.com/89315525/215777118-5987580d-b740-46b0-95ff-5433c508f78e.png)

Mas acho que a forma como você indicou é a mais correta.